src/metaschema.json

```diff
@@ -712,16 +727,57 @@
             "additional_columns": {
               "type": "string"
             },
-            "index_columns": {"type": "array", "items": {"type": "string"}},
+            "index_columns": { "type": "array", "items": { "type": "string" } },
             "initial_columns": {
               "type": "array",
-              "items": {"type": "string"}
+              "items": { "type": "string" }
             }
           },
           "required": ["selectors", "columns"],
           "additionalProperties": false
         }
       }
+    },
+    "pathRule": {
+      "type": "object",
+      "properties": {
+        "level": { "enum": ["optional", "required"] },
+        "path": { "type": "string" }
+      },
+      "required": ["level", "path"],
+      "additionalProperties": false
+    },
+    "stemRule": {
+      "type": "object",
+      "properties": {
+        "level": { "enum": ["optional", "recommended", "required"] },
+        "datatypes": {
+          "type": "array",
+          "items": { "pattern": "^[a-z]+$" }
+        },
```

Comment:
The main difference with what we currently allow is this entry, which I added to enable #1672. I can move this particular insertion to that PR, if preferred.

Reply:
Actually, let me just remove it. We can have that discussion there.

```suggestion
```